contrib/completion/git-prompt.sh

```diff
@@ -229,10 +243,10 @@ __git_ps1_show_upstream ()
 		*)	    # diverged from upstream
 			upstream="|u+${count#*	}-${count%	*}" ;;
 		esac
-		if [[ -n "$count" && -n "$name" ]]; then
+		if [ -n "$count" ] && [ -n "$name" ]; then
```

Comment:
Thank you for the quick reply.

> ```
> I think we have something to
> say about local var="val" construct to help dash.
> ```

I was not aware of Documentation/CodingGuidelines, although I should have been, so thanks for pointing it out.

As far as I can tell, CodingGuidelines and my "portability guidelines" align completely for subjects which appear in both places, although each source also mentions few things which the other doesn't.

> We allowed liberal uses of bash-ism in this file, as it was
> initially written for bash anyway.  If we were rewriting the prompt
> scripts to be usable by other shells, great.  But then we'd want to
> make sure it adheres to existing coding guidelines we have.

Well, there's no bashism in it after this patchset, and the scope of non-standard allowance is pretty much identical as CodingGuidelines allows (posix, plus "local" with the same compatibility requirements), and which is also described my own "portability guidelines" commit message.

However, I didn't change style ("test" instead of [...], "then" in a new line, etc), because it wasn't broken as far as compatibility or correctness goes, and I considered it out of scope for this commit (but also wasn't aware of CodingGuidelines).

So the style in this file is the same as before, which means some divergence from CodingGuidelines.

But as far as everything else, I think it does adhere to CodingGuidelines.

Do you want me to do anything specific on this subject?

Shall I remove the "portability guidelines" part from the commit message?

Reply:
@avih unfortunately, PR comments are not bidirectional. Please follow [this guidance](https://github.com/gitgitgadget/gitgitgadget/wiki/ReplyToThis) to reply to [the email](https://lore.kernel.org/git/xmqqjzhb28yc.fsf@gitster.g).

Reply:
Oh.. oops...

Thanks for the heads up. I'll follow these links a bit later.